common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java

```diff
@@ -100,45 +104,62 @@ public void removeNpcDialogExtractionDependent(Feature feature) {
         dialogExtractionDependents.remove(feature);
     }
 
+    @SubscribeEvent
+    public void onTick(TickEvent event) {
+        long ticks = McUtils.mc().level.getGameTime();
+        if (!collectedLines.isEmpty() && ticks > chatScreenTicks + CHAT_SCREEN_TICK_DELAY) {
+            // Send the collected screen lines
+            handleIncomingChatScreen(collectedLines);
+
+            // Reset screen line collection
+            collectedLines = new ArrayList<>();
+            chatScreenTicks = 0;
+        }
+    }
+
     @SubscribeEvent(priority = EventPriority.HIGHEST)
     public void onChatReceived(ChatPacketReceivedEvent e) {
         if (e instanceof ChatPacketReceivedEvent.GameInfo) return;
 
         Component message = e.getMessage();
         StyledText styledText = StyledText.fromComponent(message);
-        String plainMessage = styledText.getString(PartStyle.StyleType.NONE);
 
-        // Sometimes there is just a trailing newline; that does not
-        // make it a multiline message
-        if (styledText.contains("\n") && plainMessage.indexOf('\n') != plainMessage.length() - 1) {
+        long currentTicks = McUtils.mc().level.getGameTime();
+        // It is a multi-line screen if it contains a newline, or if it is empty and sent in the same tick (with some
+        // fuzziness) as the current screen
+        if (styledText.contains("\n")
+                || (styledText.isEmpty() && (currentTicks <= chatScreenTicks + CHAT_SCREEN_TICK_DELAY))) {
             // This is a "chat screen"
             if (shouldSeparateNPC()) {
-                handleIncomingChatScreen(message);
-                e.setCanceled(true);
-            }
-            // If we are not separating NPCs, just pass the chat screen right on. We can do
-            // nothing about it.
-            // FIXME: This is a temporary solution. We don't classify NPC messages correctly, they are marked as INFO.
-            //        Even if we don't separate NPC messages, we should still classify them correctly.
-            //        However, this hack allows us to disable the NPC dialog extraction feature (/overlay), and still
-            // get
-            //        the dialog in the chat window.
```

Comment:
We still don't classify NPC messages correctly. I agree this is not the best place for the FIXME, but I think we need to have it somewhere

Reply:
How do you think we should classify the chat screen messages then? As a new SCREEN type?

Reply:
The examples you sent are classified correctly, yes. Other types are just "INFO". I guess that's fine, but a bit weird.

Reply:
I think we could just classify them as NPC, even if we don't "process" them further. But it might not be reliable.

Reply:
I still don't get it, but let's take it in Discord instead.